Zend/zend_operators.c

```diff
@@ -593,20 +593,15 @@ ZEND_API void ZEND_FASTCALL convert_to_array(zval *op) /* {{{ */
 							!ZEND_HASH_GET_APPLY_COUNT(Z_OBJ_P(op)->properties)) {
 							/* fast copy */
 							if (EXPECTED(Z_OBJ_P(op)->handlers == &std_object_handlers)) {
-								arr = obj_ht;
-								if (EXPECTED(!(GC_FLAGS(Z_OBJ_P(op)->properties) & IS_ARRAY_IMMUTABLE))) {
-									GC_REFCOUNT(Z_OBJ_P(op)->properties)++;
-								}
+								arr = zend_proptable_to_symtable(obj_ht, 0);
 							} else {
-								arr = zend_array_dup(obj_ht);
+								arr = zend_proptable_to_symtable(obj_ht, 1);
 							}
-							zval_dtor(op);
-							ZVAL_ARR(op, arr);
 						} else {
-							arr = zend_array_dup(obj_ht);
-							zval_dtor(op);
-							ZVAL_ARR(op, arr);
+							arr = zend_proptable_to_symtable(obj_ht, 1);
```

Comment:
Would be nice to combine these two cases, the extra split for one out for conditions is odd.